system/Debug/Toolbar/Views/toolbar.js

```diff
@@ -421,161 +438,230 @@ var ciDebugBar = {
-            ciDebugBar.addClass(ciDebugBar.toolbarContainer, 'dark');
+        if (ciDebugBar.readCookie("debug-bar-theme") === "dark") {
+            ciDebugBar.removeClass(ciDebugBar.toolbarContainer, "light");
+            ciDebugBar.addClass(ciDebugBar.toolbarContainer, "dark");
+        } else if (ciDebugBar.readCookie("debug-bar-theme") === "light") {
+            ciDebugBar.removeClass(ciDebugBar.toolbarContainer, "dark");
+            ciDebugBar.addClass(ciDebugBar.toolbarContainer, "light");
         }
-        else if (ciDebugBar.readCookie('debug-bar-theme') === 'light')
-        {
-            ciDebugBar.removeClass(ciDebugBar.toolbarContainer, 'dark');
-            ciDebugBar.addClass(ciDebugBar.toolbarContainer, 'light');
+
+        btnTheme.addEventListener(
+            "click",
+            function () {
+                var theme = ciDebugBar.readCookie("debug-bar-theme");
+
+                if (
+                    !theme &&
+                    window.matchMedia("(prefers-color-scheme: dark)").matches
+                ) {
+                    // If there is no cookie, and "prefers-color-scheme" is set to "dark"
+                    // It means that the user wants to switch to light mode
+                    ciDebugBar.createCookie("debug-bar-theme", "light", 365);
+                    ciDebugBar.removeClass(ciDebugBar.toolbarContainer, "dark");
+                    ciDebugBar.addClass(ciDebugBar.toolbarContainer, "light");
+                } else {
+                    if (theme === "dark") {
+                        ciDebugBar.createCookie(
+                            "debug-bar-theme",
+                            "light",
+                            365
+                        );
+                        ciDebugBar.removeClass(
+                            ciDebugBar.toolbarContainer,
+                            "dark"
+                        );
+                        ciDebugBar.addClass(
+                            ciDebugBar.toolbarContainer,
+                            "light"
+                        );
+                    } else {
+                        // In any other cases: if there is no cookie, or the cookie is set to
+                        // "light", or the "prefers-color-scheme" is "light"...
+                        ciDebugBar.createCookie("debug-bar-theme", "dark", 365);
+                        ciDebugBar.removeClass(
+                            ciDebugBar.toolbarContainer,
+                            "light"
+                        );
+                        ciDebugBar.addClass(
+                            ciDebugBar.toolbarContainer,
+                            "dark"
+                        );
+                    }
+                }
+            },
+            true
+        );
+    },
+
+    setHotReloadState: function () {
+        var btn = document.getElementById("debug-hot-reload").parentNode;
+        var btnImg = btn.firstElementChild;
```

Comment:
```suggestion
        var btnImg = btn.getElementsByTagName('img')[0];
```
If we rotate the image and not the button, we will avoid the problem of rotating the padding that is assigned to the button.

Reply:
Thanks! Done.